modules/system/checks.nix

```diff
@@ -3,6 +3,9 @@
 with lib;
 
 let
+  # Similar to lib.escapeShellArg but escapes "s instead of 's, to allow for parameter expansion in shells
+  escapeDoubleQuote = arg: ''"${replaceStrings ["\""] ["\"\\\"\""] (toString arg)}"'';
+
```

Comment:
I'm not sure if this would be the best place for a utility function like this?

Reply:
Adding it here is fine as it's currently only used here